nni/algorithms/compression/v2/pytorch/pruning/tools/data_collector.py

```diff
@@ -6,13 +6,16 @@
 
 from torch import Tensor
 
-from .base import DataCollector, TrainerBasedDataCollector
+from .base import DataCollector, EvaluatorBasedDataCollector
+from .base import TrainerBasedDataCollector
 
 _logger = logging.getLogger(__name__)
 
-__all__ = ['WeightDataCollector', 'WeightTrainerBasedDataCollector', 'SingleHookTrainerBasedDataCollector']
+__all__ = ['TargetDataCollector', 'EvaluatorBasedTargetDataCollector', 'EvaluatorBasedHookDataCollector',
+           'WeightDataCollector', 'WeightTrainerBasedDataCollector', 'SingleHookTrainerBasedDataCollector']  # TODO: remove in nni v3.0.
 
 
+# TODO: remove in nni v3.0.
```

Comment:
Please throw a deprecation warning if users use this.

Reply:
most user will not use these classes, they are pruner utils, I think pruner level deprecation warning is enough, or we will have too many warning if user use the old Pruner API.